packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts

```diff
@@ -6,75 +6,123 @@ import {
     formatEthereumAddress,
     isChainIdValid,
 } from '@masknet/web3-shared'
+import { groupBy } from 'lodash-es'
 import { Flags } from '../../../utils'
 
+const NATIVE_TOKEN_ADDRESS_IN_1INCH = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee'
+
+interface Token {
+    address: string
+    chainId: number
+    name: string
+    symbol: string
+    decimals: number
+    logoURI?: string
+}
+
 interface TokenList {
     keywords: string[]
     logoURI: string
     name: string
     timestamp: string
-    tokens: {
-        address: string
-        chainId: number
-        name: string
-        symbol: string
-        decimals: number
-        logoURI?: string
-    }[]
+    tokens: Token[]
     version: {
         major: number
         minor: number
         patch: number
     }
 }
 
+interface TokenObject {
+    tokens: {
+        [key: string]: Token
+    }
+}
+
 const fetchTokenList = memoizePromise(
     async (url: string) => {
         const response = await fetch(url, { cache: 'force-cache' })
-        return response.json() as Promise<TokenList>
+        return response.json() as Promise<TokenList | TokenObject>
     },
     (url) => url,
 )
 
 /**
- * Fetch tokens from token list
+ * Fetch tokens from 1inch token list
  * @param url
  * @param chainId
  */
-export async function fetchERC20TokensFromTokenList(
+async function fetch1inchERC20TokensFromTokenList(
     url: string,
     chainId = ChainId.Mainnet,
 ): Promise<ERC20TokenDetailed[]> {
-    return (await fetchTokenList(url)).tokens
-        .filter((x) => x.chainId === chainId && isChainIdValid(chainId, Flags.wallet_allow_testnet))
+    const tokens = ((await fetchTokenList(url)) as TokenObject).tokens
+    const _tokens = Object.values(tokens)
+    return _tokens
+        .filter((x) => x.address.toLowerCase() !== NATIVE_TOKEN_ADDRESS_IN_1INCH)
```

Comment:
You can filter the ether token out in the `fetchCommonERC20TokensFromTokenList`. There is no reason an ERC20 token will use the address `0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee`.

Reply:
The 1inch token have different structure: 
```javascript
'adddress': {
  address: '',
  logoURI: '',
  ...ohter,
}
```
So, I need new method to handle this.

Reply:
It tells that the 1inch token list did not match the spec. 🤔

Reply:
I see. It was using the 1inch token list API instead.